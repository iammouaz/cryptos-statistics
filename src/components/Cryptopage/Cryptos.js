import React from 'react';
import Col from 'react-bootstrap/Col';
import Card from 'react-bootstrap/Card';
import { BsArrowRightCircle } from 'react-icons/bs';

const Crypto = (data) => {
  const {
    id, symbol, name, price, img,
  } = data;
  return (
    <Col>
      <Card>
        <BsArrowRightCircle className="align-self-end" />
        <Card.Img variant="top" src={img} className="align-self-center" />
        <Card.Body>
          <Card.Title>
            <a href={`/${id}`} className="stretched-link">
              {name}
              {` (${symbol})`}
            </a>
          </Card.Title>
          <Card.Text>
            Price: $
            {price}
          </Card.Text>
        </Card.Body>
      </Card>
    </Col>
  );
};

export default Crypto;
