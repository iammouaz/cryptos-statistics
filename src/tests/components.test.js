import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import store from '../redux/StoreConfig';
import CryptoReducer from '../redux/api/CryptoData';
// Components
import Header from '../components/header';
import Cryptos from '../components/Cryptopage/Cryptos';

describe('Components Matches', () => {
  it('Mission Components Matches', () => {
    const header = renderer
      .create(<BrowserRouter><Header /></BrowserRouter>)
      .toJSON();
    expect(header).toMatchSnapshot();
  });
  it('Cryptos Components Matches', () => {
    const cryptos = renderer
      .create(<Provider store={store}><Cryptos /></Provider>)
      .toJSON();
    expect(cryptos).toMatchSnapshot();
  });
  test('Check the initialState of the reducer', () => {
    expect(CryptoReducer(undefined, {})).toEqual([]);
  });
});
