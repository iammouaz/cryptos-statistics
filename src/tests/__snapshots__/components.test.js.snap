// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components Matches Cryptos Components Matches 1`] = `
<div
  className="MuiPaper-root MuiPaper-outlined MuiPaper-rounded MuiCard-root scard flex css-bbgsq2-MuiPaper-root-MuiCard-root"
>
  <div
    className="MuiCardContent-root css-46bh2p-MuiCardContent-root"
  >
    <div
      className="MuiTypography-root MuiTypography-h5 css-ag7rrr-MuiTypography-root"
    />
    <p
      className="MuiTypography-root MuiTypography-body2 css-r40f8v-MuiTypography-root"
      x={
        Object {
          "mb": 1.5,
        }
      }
    >
      Price: $
    </p>
  </div>
  <div
    className="MuiCardActions-root MuiCardActions-spacing css-1t6e9jv-MuiCardActions-root"
    style={
      Object {
        "flexDirection": "column",
      }
    }
  >
    <img
      alt="icon"
      className="icons"
    />
    <a
      className="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root  css-sghohy-MuiButtonBase-root-MuiButton-root"
      href="/undefined"
      onBlur={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      style={
        Object {
          "width": "100%",
        }
      }
      tabIndex={0}
    >
      See Details
    </a>
  </div>
</div>
`;

exports[`Components Matches Mission Components Matches 1`] = `
<header
  className="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionFixed flex row mui-fixed css-11b3ww9-MuiPaper-root-MuiAppBar-root"
  style={
    Object {
      "justifyContent": "space-around",
    }
  }
>
  <div
    className="flex home"
  >
    <div
      className="flex home"
    >
      <a
        aria-current="page"
        className="active"
        href="/"
        onClick={[Function]}
        style={
          Object {
            "color": "white",
            "textDecoration": "none",
          }
        }
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M10 20v-6h4v6h5v-8h3L12 3 2 12h3v8z"
          />
        </svg>
      </a>
      <div
        className="MuiTypography-root MuiTypography-h2 MuiTypography-gutterBottom css-i1hrn3-MuiTypography-root"
      >
        Home
      </div>
    </div>
  </div>
  <form
    autoComplete="off"
    className="MuiBox-root css-xn6ruc"
    id="search"
    noValidate={true}
    onChange={[Function]}
    placeholder="Search"
  >
    <div
      className="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
    >
      <label
        className="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-standard MuiFormLabel-root MuiFormLabel-colorPrimary css-aqpgxn-MuiFormLabel-root-MuiInputLabel-root"
        data-shrink={false}
        htmlFor="standard-basic"
        id="standard-basic-label"
      >
        Search
      </label>
      <div
        className="MuiInput-root MuiInput-underline MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-1480iag-MuiInputBase-root-MuiInput-root"
        onClick={[Function]}
      >
        <input
          aria-invalid={false}
          autoFocus={false}
          className="MuiInput-input MuiInputBase-input css-1x51dt5-MuiInputBase-input-MuiInput-input"
          disabled={false}
          id="standard-basic"
          onAnimationStart={[Function]}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          required={false}
          type="text"
        />
      </div>
    </div>
  </form>
</header>
`;
